data_safe_haven/infrastructure/stack_manager.py

```diff
@@ -57,20 +59,22 @@ class StackManager:
 
     def __init__(
         self,
+        context: Context,
         config: Config,
         program: DeclarativeSHM | DeclarativeSRE,
     ) -> None:
-        self.account = PulumiAccount(config)
+        self.account = PulumiAccount(context, config)
+        self.context: Context = context
         self.cfg: Config = config
```

Comment:
Not sure you need type markers here since the type is defined in the function signature.